sw/host/opentitantool/opentitantool_derive/src/lib.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use proc_macro2::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, Data, DataEnum, DeriveInput, Fields, Ident, Variant};
+
+/// Derives the `CommandDispatch` trait for a NewType enum.
+/// 
+/// Derive this on purely interior nodes in the opentitantool command hierarchy
+/// (that is: nodes whose only purpose is to dispatch down to the next layer
+/// in the command heirarchy).
+#[proc_macro_derive(CommandDispatch)]
+pub fn derive_command_dispatch(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+
+    let expanded = match &input.data {
+        Data::Enum(e) => dispatch_enum(&input.ident, e),
+        _ => unimplemented!("CommandDispatch is only implemented for enums"),
```

Comment:
I think it might be cleaner to use something like `proc_macro_error` here?

Reply:
I used `proc_macro_error` and it's `abort!` macro and manually tested the error cases.

I think that the `syn` crate may have some capabilities here in its own internal `Result` and `Error` types, but if I understood the docs correctly, I'd have to turn on a feature in the `syn` crate to use those.

I like the `proc_macro_error` well enough but I think some further investigation is needed here to understand what would be the best rust-reviewer advice when reviewing macros.